robottelo/entities.py

```diff
@@ -316,8 +337,7 @@ def create_missing(self):
         # generate an URL pointing to a docker server
         if provider.lower() == 'docker':
             if 'url' not in vars(self):
-                self.url = random.choice((
-                    get_internal_docker_url(), get_external_docker_url()))
+                self.url = self._server_config.url
```

Comment:
The ``url`` being generated here is not the same as what we expect for a docker-type repository (which is what the old code did.